backend-services/voting-app/src/main/java/org/cardano/foundation/voting/service/vote/DefaultVoteService.java

```diff
@@ -503,6 +504,102 @@ private Either<Problem, VoteEnvelope> unwrapCastCoteEnvelope(Web3ConcreteDetails
         }
     }
 
+    @Override
+    @Transactional(readOnly = true)
+    @Timed(value = "service.vote.voteReceipts", histogram = true)
+    public Either<Problem, List<VoteReceipt>> voteReceipts(JwtAuthenticationToken auth) {
+        log.info("Fetching voter's receipts for the signed data...");
+        val jwtWalletId = auth.getWalletId();
+        val jwtWalletType = auth.getWalletType();
+        val eventDetails = auth.eventDetails();
+
+        if (auth.isActionNotAllowed(VIEW_VOTE_RECEIPT)) {
+            return Either.left(Problem.builder()
+                    .withTitle("ACTION_NOT_ALLOWED")
+                    .withDetail("Action VIEW_VOTE_RECEIPT not allowed for the role:" + auth.role().name())
+                    .withStatus(BAD_REQUEST)
+                    .build());
+        }
+
+        val votes = voteRepository.findByEventIdAndWalletTypeAndWalletId(eventDetails.id(), jwtWalletType, jwtWalletId);
+        if (votes.isEmpty()) {
```

Comment:
I am not sure about this, perhaps it is simply better to return no receipts, empty array / list than an error.